### PR TITLE
fix(PodInstanceTable): fix zone and region column

### PR DESCRIPTION
**Pull request by Poltergeist**
Add the missing zone and region column to the podInstanceTable.

Deploy this config and go the detail page now the table should contain all relevant zone and Region information. as displayed in the picture below.

```JSON
{
  "id": "/with-ephemeral-vol",
  "version": "2017-01-03T17:36:39.389Z",
  "containers": [
    {
      "name": "ct1",
      "exec": {
        "command": {
          "shell": "while true; do echo the current time is $(date) > ./jdef-v1/clock; sleep 1; done"
        }
      },
      "resources": {
        "cpus": 0.1,
        "mem": 32,
        "disk": 0,
        "gpus": 0
      },
      "volumeMounts": [
        {
          "name": "v1",
          "mountPath": "jdef-v1"
        }
      ]
    },
    {
      "name": "ct2",
      "exec": {
        "command": {
          "shell": "while true; do cat ./etc/clock; sleep 1; done"
        }
      },
      "resources": {
        "cpus": 0.1,
        "mem": 32,
        "disk": 0,
        "gpus": 0
      },
      "volumeMounts": [
        {
          "name": "v1",
          "mountPath": "etc"
        }
      ]
    }
  ],
  "volumes": [
    {
      "name": "v1"
    }
  ],
  "networks": [
    {
      "mode": "host"
    }
  ],
  "scaling": {
    "kind": "fixed",
    "instances": 1,
    "maxInstances": null
  },
  "scheduling": {
    "backoff": {
      "backoff": 1,
      "backoffFactor": 1.15,
      "maxLaunchDelay": 3600
    },
    "upgrade": {
      "minimumHealthCapacity": 1,
      "maximumOverCapacity": 1
    },
    "killSelection": "Youngest_First",
    "unreachableStrategy": {
      "inactiveAfterSeconds": 900,
      "expungeAfterSeconds": 604800
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  }
}

```

image

Closes DCOS-20247

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
Alfred test this please :)

**Comment by DanielMSchmidt** [comment]
You can literally never see the local completely, but I guess this is more of a general table styling issue. I added it to the respective JIRA issue
bildschirmfoto 2018-01-17 um 15 59 33

File: plugins/services/src/js/containers/pod-instances/PodInstancesTable.js [hunk 1]
```diff
@@ -443,6 +464,34 @@ class PodInstancesTable extends React.Component {
     );
   }
 
+  renderRegion(prop, instance, rowOptions) {
+    if (rowOptions.isParent) {
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: One question out of interest, is there a reason you use this inverse guard clause pattern here?
I am used to seeing early returns in cases like this, e.g.

```js
if (!rowOptions.isParent) {
  return null;
}

return (...)
```

File: plugins/services/src/js/utils/__tests__/InstanceUtil-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  describe("#getRegionName", function() {
+    beforeEach(function() {
+      CompositeState.getNodesList = () => {
+        return new NodesList({ items: SlaveNodes });
+      };
+      CompositeState.getMasterNode = () => {
+        return new Node(MasterNodeLocal);
+      };
+    });
+    it("returns N/A when no region name exists", function() {
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/utils/__tests__/InstanceUtil-test.js]: Can we add a small test that it doesn't horribly crash when called with null? As we have no type system I would feel a bit safer with that 🙈

File: plugins/services/src/js/containers/pod-instances/PodInstancesTable.js [hunk 3]
```diff
@@ -146,6 +153,20 @@ class PodInstancesTable extends React.Component {
         render: this.renderColumnAddress,
         sortable: true
       },
+      {
```
Comment (DanielMSchmidt) [inline plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: Can we add a small test in the `PodInstancesTable-test.js` to check that these are set?

File: plugins/services/src/js/utils/__tests__/InstanceUtil-test.js (hunk 2 above)
Comment (Poltergeist) [reply plugins/services/src/js/utils/__tests__/InstanceUtil-test.js]: done

File: plugins/services/src/js/containers/pod-instances/PodInstancesTable.js (hunk 1 above)
Comment (Poltergeist) [reply plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: you are right I was just misguided.

File: plugins/services/src/js/containers/pod-instances/PodInstancesTable.js (hunk 3 above)
Comment (Poltergeist) [reply plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: done

File: plugins/services/src/js/containers/pod-instances/PodInstancesTable.js (hunk 1 above)
Comment (weblancaster) [reply plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: Would be nice to have in the rest too @Poltergeist

Comment (Poltergeist) [reply plugins/services/src/js/containers/pod-instances/PodInstancesTable.js]: I have addressed the issues from @DanielMSchmidt unfortunately forgot to push them.

**Comment by Poltergeist** [comment]
Alfred you can't test this. 👯 👯‍♂️ 🎉

**Comment by DanielMSchmidt** [comment]
Alfred test this please :)

**Comment by Poltergeist** [comment]
Alfred you can't test this. 👯 👯‍♂️ 🎉

**Comment by Poltergeist** [comment]
Alfred you can't test this. 👯 👯‍♂️ 🎉

**Comment by Poltergeist** [comment]
run integration tests

**Comment by Fabs** [comment]
Looking at this one.

**Comment by bstavroulakis** [comment]
@Poltergeist could you please rebase this with master? I've merged the tests fix and they should pass now.

**Comment by Fabs** [comment]
Just did the rebase. @DanielMSchmidt, we need your review again :-)

**Comment by DanielMSchmidt** [comment]
Alfred test this please :)

**Comment by bstavroulakis** [comment]
Alfred test this
